src/cpp/rtps/transport/shared_mem/SHMLocator.hpp

```diff
@@ -50,8 +51,17 @@ class SHMLocator
         locator.get_address()[0] = (type == Type::UNICAST) ? 'U' : 'M';
 
         auto host_id = Host::instance().id();
-        locator.get_address()[1] = octet(host_id);
-        locator.get_address()[2] = octet(host_id >> 8);
+        std::string user_name;
+
+        if ((fastdds::dds::RETCODE_OK == SystemInfo::get_username(user_name)) && (user_name.length() < 16))
+        {
+            std::memcpy(&locator.address[1], user_name.c_str(), user_name.length());
+        }
+        else
+        {
+            locator.get_address()[1] = octet(host_id);
+            locator.get_address()[2] = octet(host_id >> 8);
+        }
```

Comment:
This new approach can make a locator from the same user on a different host to be accepted.

What I would do is having the following format for the locator:

<kind> | <host_id> | <effective_user_id>

Where the first two fields are what we currently have, and the following one would be the result of `geteuid()` on POSIX systems, and something equivalent on Win32

Reply:
Additional note: we followed and discarded a more convoluted approach on #1801